SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialVirtualWorkerTranslationManager.cpp

```diff
@@ -0,0 +1,202 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "EngineClasses/SpatialVirtualWorkerTranslator.h"
+#include "Interop/Connection/SpatialWorkerConnection.h"
+#include "Interop/SpatialReceiver.h"
+#include "SpatialConstants.h"
+#include "Utils/SchemaUtils.h"
+
+DEFINE_LOG_CATEGORY(LogSpatialVirtualWorkerTranslationManager);
+
+SpatialVirtualWorkerTranslationManager::SpatialVirtualWorkerTranslationManager()
+	: bWorkerEntityQueryInFlight(false)
+{}
+
+void SpatialVirtualWorkerTranslationManager::Init(
+	USpatialReceiver* InReceiver,
+	USpatialWorkerConnection* InConnection,
+	SpatialVirtualWorkerTranslator* InTranslator)
+{
+	check(InReceiver != nullptr);
+	Receiver = InReceiver;
+
+	check(InConnection != nullptr);
```

Comment:
Looking ahead, remove this check() might make testing the `SpatialVirtualWorkerTranslationManager` easier. Reason being is that we can pass in a `nullptr` here for the InConnection, alongside a mocked translator that the translator uses. Also, we currently also check() before every usage of the connection in code so we shouldn't lose any safety?

Reply:
I moved it to a constructor. The only thing not checked was the translator, so I added a check for that.